sklearn/utils/testing.py

```diff
@@ -56,7 +56,19 @@
     raises = deprecated(deprecation_message)(_nose_raises)
 except ImportError:
     pass
-from nose import with_setup
+
+try:
+    from nose.tools import with_setup as _with_setup
+    deprecation_message = (
+        'sklearn.utils.testing.with_setup has been deprecated in version 0.20 '
+        'and will be removed in 0.22.'
+        'If your code relies on with_setup, please use'
+        ' nose.tools.with_setup instead.')
+    with_setup = deprecated(deprecation_message)(_with_setup)
+    additional_symbols_to_export = ['raises', 'with_setup']
```

Comment:
What's the story with raises?

Reply:
> What's the story with raises?

Not sure what you mean.

Reply:
Ahh now I see it deprecated above. No worries